### DISCUSSION TITLE
Update item data (patch) but nothing happens

### DISCUSSION THREAD

**Discussion by nepgya2803** [pnp/pnpcore#1253] [Q&A] [answered]
I'm new to pnp core. My goal is to update the item data (user data), after going through many documents, threads and discussion, here is what I got. Even though the debug console show that a patch request was sent, but the item still hasn't been changed
```
public async Task Update(PnPContext context)
        {
            var myList = await context.Web.Lists.GetByTitleAsync("My list");

            string viewXml = CAMLQueryGenerator("5106686");

            await myList.LoadListDataAsStreamAsync(new RenderListDataOptions()
            {
                ViewXml = viewXml,
                RenderOptions = RenderListDataOptionsFlags.ListData
            });

            foreach (var item in myList.Items)
            {
                // Here if the employee number is equal to the one need to be updated, it will do the update and stop
                if (item["EmployeeNumber"].ToString() == "5106686")
                {
                    item["Specialty"] = "xxxxxx";
                    await myList.UpdateAsync();
                    break;
                }
            }
        }
```

And here is my calm query 
```
private string CAMLQueryGenerator(string employeeID)
        {
            string caml = "<Query>" +
                    "<Where>" +
                      "<Eq>" +
                          "<FieldRef Name='EmployeeNumber'/>" +
                          "<Value Type='Text'>" + employeeID + "</Value>" +
                      "</Eq>" +
                      "</Where>" +
                    "</Query>";

            return caml;
        }
```

**Comment by jansenbe** [comment, marked as answer]
@ngqtrieu99 : that code should work although please fix the following by adding an `.AsRequested()` here:

```csharp
...
foreach (var item in myList.Items.AsRequested())
...
```

If that's not working check the payload of the patch request to ensure the field in included. Alternatively try the `UpdateOverwriteVersionAsync` (see https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#updating-list-items) method

**Reply by nepgya2803** [reply]
It really works. Changing into AsRequested() and using UpdateOverwriteVersionAsync () really did the job. Can you explain why we need them?

**Reply by jansenbe** [reply]
@ngqtrieu99 : using the `LoadListDataAsStreamAsync` you already loaded the items in the collection, when you want to iterate over the loaded items you need to add `.AsRequested()` as otherwise our LINQ provider thinks you want to perform a new query. The `UpdateOverwriteVersionAsync` uses another update model, but still that shouldn't be required. You can use Fiddler to check what goes over the wire
